dna-tools/dna-writer.cpp

```diff
@@ -72,10 +72,15 @@ int main(int argc, char *argv[])
 	dna_header[SZG_DNA_PTR_DNA_MAJOR] = DNA_SPEC_MAJOR;
 	dna_header[SZG_DNA_PTR_DNA_MINOR] = DNA_SPEC_MINOR;
 
-	// As of now there are no incompatibilities, so any dna parser should
+	// TXR4 peripherals require version 1.1, otherwise any dna parser should
 	// work.
-	dna_header[SZG_DNA_PTR_DNA_REQUIRED_MAJOR] = 0;
-	dna_header[SZG_DNA_PTR_DNA_REQUIRED_MINOR] = 0;
+	if (dna_json["is_txr4"].get<bool>() == true) {
```

Comment:
For this sort of thing, a progressive test might work better...

1. Setup the sensible default.
2. If (condition1) then add a new requirement.
3. If (condition2) then add a new requirement.

Things get complicated to cover all the tasks, so we can just go with what you have here and make it more complex later.

1. Setup the sensible default.
2. If (condition1) then assure rev >= 1.1
3. If (condition2) then assure rev >= 1.4
4. If condition3) then assure rev > 1.2
5. etc.

Reply:
I've updated the method I'm using for this in the latest commit (4c359044fd), if that looks good to you I'll merge this all.

Reply:
Looks good to me.